File: api/src/application/http/prompt/validators.rs
use serde::{Deserialize, Serialize};
use utoipa::ToSchema;
use validator::Validate;

#[derive(Debug, Serialize, Deserialize, Validate, ToSchema)]
pub struct CreatePromptValidator {
    #[validate(length(min = 1, message = "name is required"))]
    pub name: String,

    #[validate(length(min = 1, message = "description is required"))]
    pub description: String,

    #[validate(length(min = 1, message = "template is required"))]
    pub template: String,

    #[validate(length(min = 1, message = "version is required"))]
    pub version: String,
}

#[derive(Debug, Serialize, Deserialize, Validate, ToSchema)]
pub struct UpdatePromptValidator {
    #[serde(default)]
    pub name: Option<String>,

    #[serde(default)]
    pub description: Option<String>,

    #[serde(default)]
    pub template: Option<String>,

    #[serde(default)]
    pub version: Option<String>,

    #[serde(default)]
    pub is_active: Option<bool>,
}

File: front/src/pages/realm/validators.ts
import { SigningAlgorithm } from '@/api/core.interface'
import { z } from 'zod'

export const updateRealmValidator = z.object({
  name: z.string().min(1),
  default_signing_algorithm: z.nativeEnum(SigningAlgorithm),
})

export const createWebhookValidator = z.object({
  name: z.string(),
  description: z.string().optional(),
  endpoint: z.string().url().optional(),
  subscribers: z.array(z.string()),
})

export const createPromptValidator = z.object({
  name: z.string().min(1, 'Name is required'),
  description: z.string().min(1, 'Description is required'),
  template: z.string().min(1, 'Template is required'),
  version: z.string().min(1, 'Version is required'),
})

export const updatePromptValidator = z.object({
  name: z.string().min(1).optional(),
  description: z.string().min(1).optional(),
  template: z.string().min(1).optional(),
  version: z.string().min(1).optional(),
  is_active: z.boolean().optional(),
})

export type UpdateRealmSchema = z.infer<typeof updateRealmValidator>
export type CreateWebhookSchema = z.infer<typeof createWebhookValidator>
export type CreatePromptSchema = z.infer<typeof createPromptValidator>
export type UpdatePromptSchema = z.infer<typeof updatePromptValidator>
